pkg/ansible/paramconv/paramconv.go

```diff
@@ -70,6 +70,49 @@ func ToCamel(s string) string {
 	return ret
 }
 
+// This function modifies the string to handle special words that are a part of wordMapping
+/* In this function we process the input string and find the occurrence of the values in wordMapping
+   and we append "_" on both sides of that word so that it is considered to be one word*/
+// We even handle plurals of the values in wordMapping
+// If we find "s" i.e plural, we capitalize it so that it is considered to be a part of same abbreviation
+
+func preprocessWordMapping(s string) string {
+	var x int
+	var y string
+
+	for _, v := range wordMapping {
+		x = strings.Index(s, v)
+		y = v
+		if x >= 0 {
+			break
+		}
+	}
+	if x == -1 {
+		return s
+	}
+
+	if (x + len(y) - 1) == len(s)-1 {
```

Comment:
I'd like to see some logic comments here too. It isn't clear what any of these conditions are doing.